lib/image-stream/archive.js

```diff
@@ -81,8 +80,20 @@ const extractEntryByPath = (archive, filePath, options) => {
     return Bluebird.resolve(options.default);
   }
 
-  return options.hooks.extractFile(archive, options.entries, fileEntry.name)
-    .then(rindle.extract);
```

Comment:
Is the functionality that you're replacing here (and in the unit tests), generic enough to be moved to e.g. the `utils` module?